### ISSUE TITLE
[Question] for nvidia shield users

### ISSUE THREAD

**Issue by D4anielCB** [elgatito/plugin.video.elementum#742]
I'd like to know if its possible to use external drive to download the torrents in the nvidia shield directly

**Comment by antonsoroko** [comment]
can't say about nvidia shield in particular, but for arbitrary android tv box the approach is to format external HDD as internal storage and move apps data to it (necessarily).
see https://github.com/elgatito/plugin.video.elementum/issues/283#issuecomment-451776413

and then set download folder to something like `Android/data/org.xbmc.kodi/elementum/` (full path would look like `/storage/emulated/0/Android/data/org.xbmc.kodi/elementum/`) because of security constraints.


if by "directly" you mean without formatting - then no, android security restriction (at least it used to be).

**Comment by D4anielCB** [comment]
> can't say about nvidia shield in particular, but for arbitrary android tv box the approach is to format external HDD as internal storage and move apps data to it (necessarily).
> see [#283 (comment)](https://github.com/elgatito/plugin.video.elementum/issues/283#issuecomment-451776413)
> 
> and then set download folder to something like `Android/data/org.xbmc.kodi/elementum/` (full path would look like `/storage/emulated/0/Android/data/org.xbmc.kodi/elementum/`) because of security constraints.
> 
> if by "directly" you mean without formatting - then no, android security restriction (at least it used to be).

I use actually a mibox and it has a usb por 2.0 formating like you said it makes the box slow due the 2.0 speed, i never tried using like that, but as nvidia shield has a 3.0 port it works good? It has to be a external ssd?
I got the box so im deciding if i keep or return cuz the box is freaking expensinve here, as one of the reasons I want to be able to download torrents using the box. Thank you very much for replying and im sorry abt my english

**Comment by antonsoroko** [comment]
So, I use mi box 3, it has usb 2.0 (but my hdd has 3.0). And it works fine for me.
When I tried with usb stick - then yes, it was extremely slow.
But with hdd I don't feel any speed issues.

But nvidia shield might have some security quirks, so I suggest you to test approach I described above first.

Also note that android 10 is currently not supported by elementum (see relevant issue).

**Comment by antonsoroko** [comment]
also you can do some simple math to estimate your use case:

let's say for DBRemux with size 60GB and length 90min we got about 11MB per second. (of course your internet also should support that).

then theoretical speed of usd 2.0 is 480 Mb/s => 60 MB/s. in reality (on PC) it is about 40 MB/s. Which should be enough.
of course you should have at least decent HDD.

the best way to check is to install some hdd speed test application on android and see real results. then you can find out what kind of releases your hdd will be able to handle.

**Comment by D4anielCB** [comment]
> also you can do some simple math to estimate your use case:
> 
> let's say for DBRemux with size 60GB and length 90min we got about 11MB per second. (of course your internet also should support that).
> 
> then theoretical speed of usd 2.0 is 480 Mb/s => 60 MB/s. in reality (on PC) it is about 40 MB/s. Which should be enough.
> of course you should have at least decent HDD.
> 
> the best way to check is to install some hdd speed test application on android and see real results. then you can find out what kind of releases your hdd will be able to handle.

Thank you very much for all the info, it has been very useful. I only tried using as internal storage a 3.0 usb stick that i found recomended in a facebook group, never tired any hdd, i have 2 hdd 3.0 im gonna try them.
When you turn off the mibox the hdd turn off too?

**Comment by antonsoroko** [comment]
>When you turn off the mibox the hdd turn off too?

if by "turn off" you mean sleep (since there is no "turn off" in android tv) - then yes, hdd also goes to sleep.
but i also exit kodi before sleep (iirc i had some issues with hdd sleep without exit from kodi).

**Comment by D4anielCB** [comment]
> > When you turn off the mibox the hdd turn off too?
> 
> if by "turn off" you mean sleep (since there is no "turn off" in android tv) - then yes, hdd also goes to sleep.
> but i also exit kodi before sleep (iirc i had some issues with hdd sleep without exit from kodi).

yes, this is what i meant, sorry, putting it in sleep mode. So I need to close kodi every time I stop using? sometimes the exit from kodi doesnt work, it get stucked in a black screen or something =x

**Comment by antonsoroko** [comment]
i mean, you better to test it by yourself on your system. it may vary.
i do this by force of habit :-)

btw, if it got stuck - you can stop it in android setting.

also iirc i changed settings of hdd to go to sleep earlier than it was by default (WD has utility for their drives, for example).

**Comment by D4anielCB** [comment]
> i mean, you better to test it by yourself on your system. it may vary.
> i do this by force of habit :-)
> 
> btw, if it got stuck - you can stop it in android setting.
> 
> also iirc i changed settings of hdd to go to sleep earlier than it was by default (WD has utility for their drives, for example).

Yes, it can be stopped in android settings, one time I erased data instead by mistake, fortunelly i make backups of my kodi
I'll try with my hdds and probably return the nvidia (havent opened the box yet)
Thanks again, you have been kind

**Comment by D4anielCB** [comment]
Your HDD has exernal power source?

**Comment by antonsoroko** [comment]
no.

i have Western Digital My Passport USB3.0 USB 3.1 (WDBEMM0010BSL 1 TB), you can see specs here

https://translate.google.com/translate?sl=ru&tl=en&u=https://www.nix.ru/autocatalog/external_hdd_western_digital/WD-WDBEMM0010BSL-EEUE-My-Passport-USB30-Drive-1TB-Silver-25-EXT-RTL_139262.html

**Comment by D4anielCB** [comment]
> no.
> 
> i have Western Digital My Passport USB3.0 USB 3.1 (WDBEMM0010BSL 1 TB), you can see specs here
> 
> https://translate.google.com/translate?sl=ru&tl=en&u=https://www.nix.ru/autocatalog/external_hdd_western_digital/WD-WDBEMM0010BSL-EEUE-My-Passport-USB30-Drive-1TB-Silver-25-EXT-RTL_139262.html

Ah ok, thanks

**Comment by D4anielCB** [comment]
Can you name any good app to run the hdd speed test?

**Comment by antonsoroko** [comment]
for android tv - not really.
you can try https://play.google.com/store/apps/details?id=mark.fakedevicetest (i used it for phone)
or something from https://play.google.com/store/search?q=hdd%20speed&c=apps
and i guess you will need to download apk and install it manually, since these tools probably will be unavailable in android TV play marker.

**Comment by D4anielCB** [comment]
That will be no problem, thanks

**Comment by D4anielCB** [comment]
> for android tv - not really.
> you can try https://play.google.com/store/apps/details?id=mark.fakedevicetest (i used it for phone)
> or something from https://play.google.com/store/search?q=hdd%20speed&c=apps
> and i guess you will need to download apk and install it manually, since these tools probably will be unavailable in android TV play marker.

I have a thumb drive 3.0 with 64gb I tested it and it was around 6mb/s writing speed, i decided to test it and it worked but sometimes it was a bit laggy. I saw a lot of people were recomeding thje sandisk SDCZ430 so i decided to buy and its around 17mb/s writing speed and its working perfectly. I don't really need much space just to alocate the actual torrent im watching i nthe moment, i don't keep them.
This is just my feedback and again thank you for sharing your experience with me, it was really helpful

**Comment by D4anielCB** [comment]
@elgatito if you want to close this, maybe it will be helpful for others. Thx

**Comment by antonsoroko** [comment]
I would not recommend to use USB stick, since they all very slow (in comparison to hdd) unless you use some ssd nvme usb stick (which will be unreasonably expensive for this use case).
And with HDD you can contribute to community by keeping and seeding files.
Anyway, in your case you also can try yo use "memory storage" option if you don't need to download something beforehand and watch later. This probably will be more smooth experience than with usb stick.

**Comment by antonsoroko** [comment]
@D4anielCB could you please rename this issue, e.g. "How to use external HDD with elementum on android"

**Comment by antonsoroko** [comment]
@D4anielCB ?
